lib/stack_master/parameter_resolvers/ami_finder.rb

```diff
@@ -0,0 +1,43 @@
+module StackMaster
+  module ParameterResolvers
+    class AmiFinder
+      def initialize(region)
+        @region = region
+        @owner = [aws_account_id]
+      end
+
+      def build_filters(value, prefix = nil)
+        value.split(',').map do |name_with_value|
+          name, value = name_with_value.strip.split('=')
+          @owner.push name if name == 'owner_id'
+          name = prefix ? "#{prefix}:#{name}" : name
+          puts value.class
+          { name: name, values: [value] }
+        end
+      end
+
+      def find_latest_ami(filters)
+        images = ec2.describe_images(owners: @owner, filters: filters).images
+        sorted_images = images.sort do |a, b|
+          Time.parse(a.creation_date) <=> Time.parse(b.creation_date)
+        end
+        sorted_images.last
+      end
+
+      private
+
+      def aws_account_id
+        arn = iam.list_users.users.first.arn
```

Comment:
Unfortunately, this wouldn't work in a scenario when the account didn't have any users.  There are certainly AWS accounts which delegate user management to other systems (via federation or role assumption).

Alas, there isn't a canonical way to find the current account's account number.

However, there _is_ a rather roundabout way to get it:
```
$ aws elastictranscoder list-presets --query 'Presets[0].Arn' | cut -d: -f5
XXXXXXXXXXXX
```

Obviously the caller needs permissions to call `elastictranscoder:listPresets` for that to work.